gcc.cmake

```diff
@@ -26,10 +26,34 @@ set(sel4_arch @KernelSel4Arch@)
 set(arch @KernelArch@)
 set(mode @KernelWordSize@)
 
+# This function hunts for an extant `gcc` with one of the candidate prefixes
+# specified in `ARGN`, allowing us to try different target triple prefixes for
+# cross-compilers built in various ways.
+function(FindPrefixedGCC out_var)
+    set("${out_var}" "PrefixedGCC-NOTFOUND")
+    foreach(prefix ${ARGN})
+        set("test_var" "_GCC_${prefix}")
+        find_program("${test_var}" "${prefix}gcc")
+        if(${test_var})
+            message(STATUS "Found GCC with prefix ${prefix}")
+            set("${out_var}" "${prefix}")
+            break()
+        endif()
+    endforeach()
+    if(${out_var})
+        set("${out_var}" "${${out_var}}" PARENT_SCOPE)
+    else()
+        message(FATAL_ERROR "Unable to find valid cross-compiling GCC")
+    endif()
+endfunction(FindPrefixedGCC)
+
 if("${CROSS_COMPILER_PREFIX}" STREQUAL "")
     if(("${arch}" STREQUAL "arm") OR ("${arch}" STREQUAL "x86") OR ("${arch}" STREQUAL "riscv"))
         if(${sel4_arch} STREQUAL "aarch32" OR ${sel4_arch} STREQUAL "arm_hyp")
-            set(CROSS_COMPILER_PREFIX "arm-linux-gnueabi-" CACHE INTERNAL "")
```

Comment:
This and the other lines changed below have the consequence of losing the `CACHE INTERNAL` with an empty string directive. I'm not entirely across how this affects `cmake -L` and if the change is a problem. I didn't put it on the final `set()` in the helper function because I don't know how `PARENT_SCOPE` and the other directives interact. Would appreciate input on what to do to make this all work happily.

Reply:
Removing the cache attribute is most probably fine as the toolchain file gets processed when the first project() statement is called and CMake requires this to occur in the first file it starts processing.  Thus the variable practically ends up with the same availability as if it was in the CACHE.